src/arangodb/arango_client.py

```
@@ -196,6 +197,30 @@ def is_connected(self) -> bool:
+            expected_type: Expected type of the result
+            default: Default value if result is not of expected type
+            
+        Returns:
+            Processed result
+        """
+        if isinstance(result, expected_type):
+            return result
+        elif result is not None and hasattr(result, 'result'):
+            # Handle AsyncJob/BatchJob
+            result_data = result.result() if callable(getattr(result, 'result', None)) else default
+            if expected_type == list and isinstance(result_data, list):
```

Ruff (0.11.9)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)